### DISCUSSION TITLE
GitHub Pages Deploy

### DISCUSSION THREAD

**Discussion by github-cooper** [vuepress/core#1426] [Q&A] [answered]
how to deploy with github pages with npm . What's the GitHub Actions file look like?

**Comment by Arhell** [comment, marked as answer]
https://v2.vuepress.vuejs.org/guide/deployment.html
or use https://github.com/peaceiris/actions-gh-pages

**Comment by tchajed** [comment]
You can use an official action for this now: https://github.com/actions/deploy-pages

actions/deploy-pages is nice because it directly creates a Pages deployment, while the older solutions used the gh-pages branch of your repo. It's also officially supported.

**Reply by Mister-Hope** [reply]
In fact I knew this official action for a long time. It doesn't make any benefits if you are suing VuePress, not only it requires an extra job, you also needs to write an extra step to deploy artifacts yourself.

I won't consider it unless preview feature for pr is stable.
